libs/wingsdk/src/target-tf-aws/api.ts

```diff
@@ -0,0 +1,217 @@
+import { Construct } from "constructs";
+import { App } from "./app";
+import { Function } from "./function";
+import { core } from "..";
+import * as cloud from "../cloud";
+import { convertBetweenHandlers } from "../convert";
+import { IResource } from "../core";
+import { Code } from "../core/inflight";
+import { NameOptions, ResourceNames } from "../utils/resource-names";
+
+/**
+ * RestApi names are alphanumeric characters, hyphens (-) and underscores (_).
+ */
+const NAME_OPTS: NameOptions = {
+  disallowedRegex: /[^a-zA-Z0-9\_\-]+/g,
+};
+
+/**
+ * AWS Implementation of `cloud.Api`.
+ */
+export class Api extends cloud.ApiBase {
+  private readonly api: WingRestApi;
+  constructor(scope: Construct, id: string, props: cloud.ApiProps = {}) {
+    super(scope, id, props);
+    this.api = new WingRestApi(this, "api");
+  }
+
+  public get(
+    route: string,
+    inflight: core.Inflight,
+    props?: cloud.ApiOnGetProps
+  ): void {
+    const fn = this.addHandler(inflight, props);
+    this.api.addEndpoint(route, ApiMethods.GET, fn);
+
+    core.Resource.addConnection({
+      from: this,
+      to: fn,
+      relationship: "on_get_request",
+    });
+  }
+
+  private addHandler(
+    inflight: core.Inflight,
+    props?: cloud.ApiOnGetProps
+  ): Function {
+    const inflightNodeHash = inflight.node.addr.slice(-8);
+
+    let fn;
+    if (
+      this.node.tryFindChild(`${this.node.id}-OnGetRequest-${inflightNodeHash}`)
+    ) {
+      fn = this.node.tryFindChild(
+        `${this.node.id}-OnGetRequest-${inflightNodeHash}`
+      );
+    } else {
+      const functionHandler = convertBetweenHandlers(
+        this,
+        `${this.node.id}-OnGetRequestHandler-${inflightNodeHash}`,
+        inflight,
+        join(__dirname, "api.ongetrequest.inflight.js"),
+        "ApiOnGetRequestHandlerClient"
+      );
+      fn = new cloud.Function(
+        this,
+        `${this.node.id}-OnGetRequest-${inflightNodeHash}`,
+        functionHandler,
+        props
+      );
+    }
+    if (!(fn instanceof Function)) {
+      throw new Error("Api only supports creating tfaws.Function right now");
+    }
+    return fn;
+  }
+  /** @internal */
+  public _bind(): void {
+    throw new Error("Method not implemented.");
+  }
+
+  /** @internal */
+  public _toInflight(): Code {
+    throw new Error("Method not implemented.");
+  }
+}
+
+/**
+ * Represents a resource with an inflight "handle" method that can be passed to
+ * `Api.get`.
+ *
+ * @inflight `wingsdk.cloud.IApiGetRequestHandler`
+ */
+export interface IApiGetRequestHandler extends IResource {}
+
+export enum ApiMethods {
+  GET = "get",
+  POST = "post",
+  PUT = "put",
+  DELETE = "delete",
+  PATCH = "patch",
+  HEAD = "head",
+  OPTIONS = "options",
+  ANY = "any",
+}
+
+export class WingRestApi extends Construct {
+  public readonly api: ApiGatewayRestApi;
+  private readonly deployment: ApiGatewayDeployment;
+  private readonly region: string;
+  private apiSpec: any = {
```

Comment:
Feels like all the OpenAPI generation stuff can go into `cloud.ApiBase` and reused across implementations.

Reply:
I will try that

Reply:
started to move it up see https://github.com/winglang/wing/pull/1285/commits/6b5141121e1cd57ce4f8145121a763bb045e76fe

The idea would be that the apiBase has a function to add a route to the apiSpec and gets some extension values from the cloud specific implementation that is then added to the route. At the end the ApiBase gives back the finalized apiSpec.